lib/mongo/topology_description.ex

```diff
@@ -123,19 +123,20 @@ defmodule Mongo.TopologyDescription do
       |> Keyword.get(:read_preference)
       |> ReadPreference.merge_defaults()
 
-    {servers, read_prefs} =
+    {server, read_prefs} =
       case topology.type do
         :unknown ->
-          {[], nil}
+          {nil, nil}
 
         :single ->
-          {topology.servers, nil}
+          server = pick_server(topology.servers)
+          {server, ReadPreference.to_topology_single_type(server)}
 
         :sharded ->
-          {mongos_servers(topology), ReadPreference.to_mongos(read_preference)}
+          {mongos_servers(topology) |> pick_server(), ReadPreference.to_mongos(read_preference)}
```

Comment:
Did you execute `mix credo`? It should complain about the |> operator.

```
topology  |> mongos_servers() |> pick_server()
```

Looks nicer.